dataflow/src/main/java/org/checkerframework/dataflow/analysis/AnalysisResult.java

```diff
@@ -195,6 +201,52 @@ private static void mergeTreeLookup(
         return merged;
     }
 
+    /**
+     * Returns true if the given tree is dead code (will never be executed at run time).
+     *
+     * <p>This method is conservative. If it returns true, the tree is dead code. If it returns
+     * false, the tree may or may not be dead code.
+     *
+     * @param tree a tree
+     * @return true if the tree is dead code
+     */
+    public boolean isDeadCode(Tree tree) {
+        if (!(tree instanceof ExpressionTree)) {
+            return false;
+        }
+        switch (tree.getKind()) {
+            case MEMBER_SELECT:
+                return false;
+            case ASSIGNMENT:
+                // This may be an expression or a statement.
```

Comment:
AssignmentTrees represent assignment expression; see their Javadoc.  `int x = 0;` is not an AssignmentTree, it's a VariableTree.

Reply:
An assignment expression `x = 0` is also a legal statement.  This not a remark about variable declarations such as `int x = 0`.  I'll remove the comment since it doesn't seem to shed light.